libraries/standard/mqtt/include/mqtt.h

```diff
@@ -53,14 +73,14 @@ typedef enum MQTTPubAckType
 
 struct MQTTTransportInterface
 {
-    TransportWriteFunc_t write;
-    TransportReadFunc_t read;
-    MQTTNetworkContext_t * networkContext;
+    MQTTTransportSendFunc_t send;
+    MQTTTransportRecvFunc_t recv;
```

Comment:
Looks like we are missing this definition.

Reply:
It's in iot_mqtt_lightweight.h

Reply:
Lightweight needs recv (to read the packet off the network) but doesn't need send. Only the full API, which transmits packets, needs send.

Reply:
I would prefer to have transport layer send receive interface declared in a separate file. It is not exactly mqtt. That way both send and receive will remain in one file.

Reply:
We discussed that, but it would make it harder to consume these libraries independently.  It would require files other than the MQTT files to be brought in to a project.